openvpn/session/authentication.go

```diff
@@ -0,0 +1,47 @@
+package session
+
+import (
+	"github.com/mysterium/node/identity"
+	"github.com/mysterium/node/openvpn/middlewares/client/auth"
+	"github.com/mysterium/node/session"
+)
+
+const sessionSignaturePrefix = "SessionId:"
+
+//NewSignedSessionIdCredentialsProvider returns session id as username and id signed with given signer as password
+func NewSignedSessionIdCredentialsProvider(id session.SessionID, signer identity.Signer) auth.CredentialsProvider {
+	signature, err := signer.Sign([]byte(sessionSignaturePrefix + id))
+
+	return func() (string, string, error) {
+		return string(id), signature.Base64(), err
+	}
+}
+
+type sessionLookupFunc func(session session.SessionID) (session.Session, bool)
+
+type verifierFactory func(identity.Identity) identity.Verifier
+
+type sessionAuthenticator struct {
+	sessionLookup  sessionLookupFunc
+	createVerifier verifierFactory
+}
+
+//NewSessionAuthenticator provides glue code for openvpn management interface to validate incoming client login request,
+//it expects session id as username, and session signature signed by client as password
+func NewSessionAuthenticator(sessionLookup sessionLookupFunc, verifierCreator verifierFactory) *sessionAuthenticator {
+	return &sessionAuthenticator{sessionLookup: sessionLookup, createVerifier: verifierCreator}
+}
+
+//ValidateSession validates provided session id and signature, by retrieving peer identity and checking session signature
+func (sa *sessionAuthenticator) ValidateSession(sessionString, signatureString string) (bool, error) {
```

Comment:
`sessionAuthenticator` struct has one method - `ValidateSession`.
We could simplify this into function type:
```go
type SessionValidator func(sessionString, signatureString string) (bool, error)
func NewSessionValidator(sessionLookup sessionLookupFunc, verifierCreator verifierFactory) SessionValidator {
...
}
```

Reply:
FunctionFactory which returns function. Fixed